static/app/components/events/breadcrumbs/utils.tsx

```diff
@@ -0,0 +1,143 @@
+import styled from '@emotion/styled';
+
+import type {SelectOption} from 'sentry/components/compactSelect';
+import type * as Timeline from 'sentry/components/timeline';
+import {
+  IconCursorArrow,
+  IconFire,
+  IconFix,
+  IconInfo,
+  IconLocation,
+  IconMobile,
+  IconRefresh,
+  IconSort,
+  IconSpan,
+  IconStack,
+  IconTerminal,
+  IconUser,
+  IconWarning,
+} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {BreadcrumbType, type RawCrumb} from 'sentry/types/breadcrumbs';
+import {toTitleCase} from 'sentry/utils/string/toTitleCase';
+
+export const BREADCRUMB_TIMESTAMP_PLACEHOLDER = '--';
+const BREADCRUMB_TITLE_PLACEHOLDER = t('Generic');
+
+export const enum BreadcrumbTimeDisplay {
+  RELATIVE = 'relative',
+  ABSOLUTE = 'absolute',
+}
+export const BREADCRUMB_TIME_DISPLAY_OPTIONS = [
+  {label: t('Relative'), value: BreadcrumbTimeDisplay.RELATIVE},
+  {label: t('Absolute'), value: BreadcrumbTimeDisplay.ABSOLUTE},
+];
+export const BREADCRUMB_TIME_DISPLAY_LOCALSTORAGE_KEY = 'event-breadcrumb-time-display';
+
+const Color = styled('span')<{colorConfig: Timeline.ColorConfig}>`
+  color: ${p => p.theme[p.colorConfig.primary]};
+`;
+
+export function applyBreadcrumbSearch(search: string, crumbs: RawCrumb[]): RawCrumb[] {
+  return crumbs.filter(
```

Comment:
if search is empty string maybe skip the filter which has that expensive JSON.stringify in it